embedded-graphics/src/primitives/triangle/mod.rs

```diff
@@ -383,6 +425,15 @@ mod tests {
         }
     }
 
+    #[test]
+    fn triangle_contains_edge_point_regression() {
+        // This test is a regression test case found while optimizing Triangle::contains()
+        assert!(
+            Triangle::new(Point::new(30, 30), Point::new(0, 0), Point::new(32, 33))
+                .contains(Point::new(31, 31))
+        );
+    }
+
     // FIXME: Colinear triangles are rendered as a line, so this should also return true. Why not?
```

Comment:
@jamwaffles what's your issue here? Comment says "should also return true" and then you assert that it returns false.